src/client/components/forms/parts/achievement.js

```diff
@@ -56,64 +56,54 @@ const maxAchievementProgress = {
 	29: 10,
 	30: 100
 };
-class Achievement extends React.Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			achievement: props.achievement,
-			counter: props.counter,
-			unlocked: props.unlocked
-		};
-	}
 
-	render() {
-		let imgElement;
-		if (this.state.unlocked) {
-			imgElement = (
-				<DragAndDropImage
-					achievementId={this.state.achievement.id}
-					achievementName={this.state.achievement.name}
-					height="100px"
-					src={this.state.achievement.badgeUrl}
-					style={{zIndex: 2}}
-				/>
-			);
-		}
-		else {
-			imgElement = (
-				<img
-					alt={this.state.achievement.name}
-					height="100px"
-					src={this.state.achievement.badgeUrl}
-					style={{zIndex: 2}}
-				/>
-			);
-		}
-		return (
-			<Card bg="light">
-				<Container fluid>
-					<Row>
-						<Col md={2}>
-							{imgElement}
-						</Col>
-						<Col md={8}>
-							<div className="h2">
-								{this.state.achievement.name}
-							</div>
-							<p>{this.state.achievement.description}</p>
-						</Col>
-						{!this.state.unlocked &&
-						<Col>
-							<div className="h3">
-								{this.state.counter}/{maxAchievementProgress[this.state.achievement.id] ?? 0}
-							</div>
-						</Col>
-						}
-					</Row>
-				</Container>
-			</Card>
-		);
-	}
+function Achievement(props) {
+	const {achievement, counter, unlocked} = props;
+	const imgElement = unlocked ? (
+		<DragAndDropImage
+			achievementId={achievement.id}
+			achievementName={achievement.name}
```

Comment:
For clarity and because we use these props everywhere in this component, what about destructuring the achievement id, name, etc. like you do for the props?

Reply:
I did think about it initially but decide to wait for your review to get idea of how you want it to be done. I will immediately work on it.